### PR TITLE
fix: [PAYG-1295] Improve error handling when parsing invalid signature

### PR DESCRIPTION
**Pull request by Couper4**
Improves error handling and readability of the error when parsing a signature in an invalid format.

### CODE REVIEW

File: csharp/test/UsageTest.cs [hunk 1]
```diff
@@ -316,6 +316,16 @@ public void Verifier_ExtractKid()
             Verifier.ExtractKid(tlSignature).Should().Be(Kid);
         }
 
+        [Fact]
```
Comment (tl-riccardo-giambona) [inline csharp/test/UsageTest.cs:319]: Should we add the same test also for `ExtractJku` and `Verify` methods?

**Review by tl-riccardo-giambona — APPROVED** [review]

**Comment by tl-wajid-malik** [comment]
Is this a C# fix only or could we port it to the other langs too?

File: .github/workflows/java.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ jobs:
       - name: Checkout current branch
         uses: actions/checkout@v3
       - name: Setup JDK
-        uses: actions/setup-java@v2
```
Comment (dili91) [inline .github/workflows/java.yml:30]: 👍

File: .github/workflows/java7.yml [hunk 3]
```diff
@@ -21,13 +21,13 @@ jobs:
       fail-fast: false
       matrix:
         os: [ ubuntu-latest ]
-        java: [ 8, 9, 10, 11, 12, 13, 14, 15, 16, 17 ]
```
Comment (dili91) [inline .github/workflows/java7.yml:24]: Makes sense, as discussed

File: java7/lib/src/test/java/com/truelayer/signing/UsageTest.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package com.truelayer.signing;
 
+import org.hamcrest.core.StringContains;
```
Comment (dili91) [inline java7/lib/src/test/java/com/truelayer/signing/UsageTest.java:3]: Is this required? For code linting you could adopt a similar approach to what we do on the BE lib, leveraging spotless. More [here](https://github.com/TrueLayer/truelayer-java#code-linting).
Anyway, I'd keep this as a separate PR if interested

**Review by dili91 — APPROVED** [review]
I've checked the Java lib and LGTM

File: php/src/Verifier.php [hunk 5]
```diff
@@ -189,8 +190,11 @@ public function requireHeaders(array $headers): IVerifier
      */
     public function verify(string $signature): void
     {
-        $jws = $this->serializerManager
-            ->unserialize($signature);
+        try {
+            $jws = $this->serializerManager->unserialize($signature);
+        } catch (Exception $e) {
+            throw new InvalidSignatureException('Failed to parse JWS: ' . $e->getMessage());
```
Comment (lboynton) [inline php/src/Verifier.php:196]: May be worth chaining the original exception here:

```suggestion
            throw new InvalidSignatureException('Failed to parse JWS: ' . $e->getMessage(), 0, $e);
```

**Comment by lboynton** [comment]
PHP side LGTM

**Review by enricocasadei — APPROVED** [review]

File: go/tlsigning_test.go [hunk 6]
```diff
@@ -102,6 +102,27 @@ func TestVerifyStaticSignature(t *testing.T) {
 	assert.Nilf(err, "signature verification should not fail: %v", err)
 }
 
+func TestVerify_WithInvalidSignature_ShouldReturnError(t *testing.T) {
```
Comment (tl-marco-tormento) [inline go/tlsigning_test.go:105]: The name of this function should be `TestVerifyWithInvalidSignatureShouldReturnError`

File: go/tlsigning_test.go [hunk 7]
```diff
@@ -102,6 +102,27 @@ func TestVerifyStaticSignature(t *testing.T) {
...
+	path := "/merchant_accounts/a61acaef-ee05-4077-92f3-25543a11bd8d/sweeping"
+
+	err := VerifyWithPem(publicKeyBytes).
+		Method("POST").
+		Path(path).
+		Header("X-Whatever-2", []byte("t2345d")).
+		Header("Idempotency-Key", idempotencyKey).
+		Body(body).
+		Verify("an-invalid..signature")
+	assert.NotNilf(err, "signature verification should not fail: %v", err)
```
[the remark this answers is not in the document]
Comment (Couper4) [reply go/tlsigning_test.go:121]: Yes, here I'm asserting that the error is not `nil`.

**Review by tl-flavio-barinas — APPROVED** [review]
python and rust LGTM
